Source/trigs.cpp

```diff
@@ -623,15 +604,14 @@ bool ForceL3Trig()
 	}
 	if (currlevel == 17) {
 		for (int i = 0; L6TWarpUpList[i] != -1; i++) {
-			if (dPiece[cursmx][cursmy] == L6TWarpUpList[i]) {
+			if (dPiece[cursPosition.x][cursPosition.y] == L6TWarpUpList[i]) {
 				for (int j = 0; j < numtrigs; j++) {
 					if (trigs[j]._tmsg == WM_DIABTWARPUP) {
-						int dx = abs(trigs[j].position.x - cursmx);
-						int dy = abs(trigs[j].position.y - cursmy);
+						int dx = abs(trigs[j].position.x - cursPosition.x);
+						int dy = abs(trigs[j].position.y - cursPosition.y);
 						if (dx < 4 && dy < 4) {

```

Comment:
I wonder if we could instead use one of the distance calculating methods build in to Point for this.

Reply:
```suggestion
						if (trigs[j].position.WalkingDistance(cursPosition) < 4) {
```

Reply:
This is fairly out of scope. I'd like to leave that for another pass if possible.

Reply:
Would have preferred it be part of the change, but it's ugly enough that someone might still consider if it can be cleaned up at a later point.